pkg/local_object_storage/shard/list.go

```diff
@@ -27,6 +28,7 @@ func (s *Shard) List() (*SelectRes, error) {
 
 	res := new(SelectRes)
 	filters := object.NewSearchFilters()
+	filters.AddPhyFilter()
```

Comment:
Change `List()` test to check this condition. I suggest to add parent for all objects in the test. They should be ignored in `List()` output. 

https://github.com/nspcc-dev/neofs-node/blob/master/pkg/local_object_storage/shard/list_test.go#L36-L50

```go
		for j := 0; j < N; j++ {
			obj := generateRawObjectWithCID(t, cid)
			addPayload(obj, 1<<2)

			// add parent as virtual object, it must be ignored in List()
			parent := generateRawObjectWithCID(t, cid)
			obj.SetParentID(parent.Object().ID())
			obj.SetParent(parent.Object().SDK())

			objs[obj.Object().Address().String()] = 0

			putPrm.WithObject(obj.Object())

			_, err := sh.Put(putPrm)
			require.NoError(t, err)
		}
	}
```